### PR TITLE
Improve delete_orphaned_versions command

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

The `get_deleted` method wouldn't work in our environment so it has been replaced with a more efficient one that gives the same results.

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions
* [ ] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/cleanup/management/commands/delete_orphaned_versions.py [hunk 1]
```diff
@@ -41,7 +43,28 @@ def handle(self, *args, **options):
...
+                'object_id',
+            ).difference(
+                all_model_ids,
+            )
+
+            """
+            The Version's `get_deleted` method was originally used here, but it was not working
+            quickly enough for our use case. The records to be deleted are now being determined
+            in a way that is easier on the database being used.
+            """
```
Comment (reupen) [inline datahub/cleanup/management/commands/delete_orphaned_versions.py]: Shouldn't this be a normal comment?

File: changelog/delete-orphaned-versions.internal.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The `delete_orphaned_versions` command was modified to select records to be deleted in a fashion that is more
```
Comment (reupen) [inline changelog/delete-orphaned-versions.internal.md]: Small thing but I would see this as either a `feature` or `bugfix` news fragment (`internal` I would take to mean that there is no change to how the app behaves).

File: datahub/cleanup/management/commands/delete_orphaned_versions.py (hunk 1 above)
Comment (elcct) [reply datahub/cleanup/management/commands/delete_orphaned_versions.py]: Yes! I updated it now. Thank you.

File: changelog/delete-orphaned-versions.internal.md (hunk 2 above)
Comment (elcct) [reply changelog/delete-orphaned-versions.internal.md]: I changed it to `bugfix`. Thank you!

**Comment by elcct** [comment]
Thank you! I moved the query creation into a separate function now.

**Pull request merged**